[18.0][IMP] web_theme_classic Add dark mode and some cleanup

ljmnoonan opened the pull request:
This could be 2 separate PRs, so please let me know if you would like me to split them.
The original purpose of this PR was to make this module play with [web_dark_mode](https://github.com/OCA/web/tree/18.0/web_dark_mode) as promised in my previous PR, #3324. This is implemented in 5f2789c465c75fccb86cc0f32bcba3b2556865df.

Here are some screenshots of this module in use in conjunction with web_dark_mode:
Before:
<img width="1920" height="1032" alt="image" src="https://github.com/user-attachments/assets/78cb3d89-f071-41cf-9fb7-a743ca20712e" />

After:
<img width="1920" height="1032" alt="image" src="https://github.com/user-attachments/assets/3fe582bd-d277-4827-b65d-92b28414ce21" />
<img width="1920" height="1032" alt="image" src="https://github.com/user-attachments/assets/221df44b-5e8e-469c-988d-cdacc966f594" />



However, while implementing the dark mode assets, I discovered a whole bunch of broken features and deficiencies in the existing code, so I decided to go about fixing them.  All the changes are detailed in the commit message on a3e37ed5d9a0bd650add143df9da8962f2f6602b, but I will attach some before and after pictures here for clarity.



## Add $wtc-input-color-required
Before:
<img width="1230" height="135" alt="image" src="https://github.com/user-attachments/assets/38e8e8fd-7cd5-4390-81f6-cfa35f444b2c" />

After:
<img width="1221" height="121" alt="image" src="https://github.com/user-attachments/assets/8019efba-212f-4b25-95e4-1635e53d7007" />
<img width="1217" height="106" alt="image" src="https://github.com/user-attachments/assets/3d967768-9b16-498e-af6e-9c5dfb635b9f" />



## Add CSS selectors to consistently handle many2many fields added as a custom property
Before:
<img width="1239" height="222" alt="image" src="https://github.com/user-attachments/assets/5db38712-d735-43c7-9bef-71ee6a534d76" />

After:
<img width="1232" height="214" alt="image" src="https://github.com/user-attachments/assets/57a93588-9729-4d7d-9f9d-6660df783fa3" />
<img width="1231" height="161" alt="image" src="https://github.com/user-attachments/assets/4d46f1d8-a274-4e50-9d36-984ee10ab342" />



## Fix monetary field selectors.
Before:
<img width="476" height="278" alt="image" src="https://github.com/user-attachments/assets/21a764f4-06c3-4fdb-a466-83ec732b811f" />
<img width="564" height="198" alt="image" src="https://github.com/user-attachments/assets/4fa603c6-9f13-43a7-877d-e3ad964d0c9d" />

After:
<img width="391" height="275" alt="image" src="https://github.com/user-attachments/assets/12eaf4a8-9b0e-431b-a3a2-70b350e77f7b" />
<img width="446" height="156" alt="image" src="https://github.com/user-attachments/assets/cc102d69-fba0-4489-99e9-645b5da86d91" />


Please let me know if you would like any changes!

ljmnoonan committed:
[IMP] web_theme_classic Refactor SCSS
Refactor all selectors to mirror the way odoo sets them. This results in a MUCH broader application of our styles that is not limited to form views.
Prefix all vars with "wtc-" in order to prevent them from overriding or being overridden by bootstrap vars.
Suffix all vars with !default to ensure that they can be overriden by dark mode.
Add $wtc-input-color-required to ensure text contrasts on $wtc-input-background-color-required. This is especially necessary in instances like the quantity and unit price columns of a sale order because by default these fields are set to a light blue.
Remove border on .note-editable, i.e. the HTML editor. This feature was broken and, as far as I am concerned, unnecessary.
Remove all button customization as it was all broken and is superceeded by OCA/web/web_save_discard_button anyways.

ljmnoonan committed:
[IMP] web_theme_classic Add dark mode assets
web_dark_mode is not merged yet.
https://github.com/OCA/web/pull/3324

legalsylvain reviewed (approved):
LGTM. Thanks for this refactoring and improvment !

remi-filament commented:
Hi @ljmnoonan thanks for this PR, I started looking at how to improve borders on monetary fields and realized you did it as well !

Would you mind trying to also fix the currency symbol which is not aligned with amount :
<img width="586" height="266" alt="image" src="https://github.com/user-attachments/assets/5aeead21-d1be-4246-8336-a72e6a11da8d" />

While testing, I found out that adding 
```css
align-self: center;
```
does the trick (although Firefox reports that it has no effect)

ljmnoonan commented:
@remi-filament, thank you very much for the heads up!
Even though Firefox doubts it, `align-self: center;` does seem to be the best fix for the the wonky placement of the monetary symbol.
While working on this I also changed the 2px margin to keep the symbol off the border to be on the symbol element instead of the input element. This allows me to account for which side it was on, although I must admit, the css selectors are a little crazy and will surely break if odoo ever changes anything, but at least it won't be very noticeable.
Also, I discovered that the searchbar in settings was getting a double border, so I fixed that.
The one remaining problem that I am noticing is that if you put invalid values into an input the background glows red as expected, but the border does not change to red. I do not think this can be realistically fixed though, so just leaving it.

Screenshots:

<img width="378" height="143" alt="image" src="https://github.com/user-attachments/assets/ac96d45a-9591-4c9e-b33c-af5026945df3" />

<img width="376" height="142" alt="image" src="https://github.com/user-attachments/assets/4e5d515e-660c-4d6a-8e93-c35afdc08ed9" />

<img width="351" height="146" alt="image" src="https://github.com/user-attachments/assets/2877a161-376d-47a6-aed9-146ade3b7777" />

<img width="319" height="72" alt="image" src="https://github.com/user-attachments/assets/fef98315-18bc-4f66-b658-ebcaa0531f6e" />

<img width="359" height="83" alt="image" src="https://github.com/user-attachments/assets/ff9efbbe-c126-474e-bcdc-645be792f52f" />

remi-filament reviewed (approved):
Great, thanks @ljmnoonan it is much better and I think current limitations are accceptable (were already there before I guess) !
Maybe it is worth adding them in README in ROADMAP section ?

legalsylvain commented:
Hi. ready to merge ? 

regards

ljmnoonan commented:
@legalsylvain 
Not yet, please. I am working on a solution for invalid fields. Should get it done today

ljmnoonan commented:
@legalsylvain @remi-filament 
This should be ready to merge now. However, I would appreciate if you look it over again. 
Long story short, while trying to get the "field invalid" styling to work I went down the rabbit hole figuring out how odoo does it. It turns out that they do not actually change the borders, but rather the color variable in css on the front end. We were using a sledgehammer by overriding it on the backend with a scss variable. Also, odoo was actually setting borders as transparent by default, but we were trying to hide them via border: 0; which caused the alignment issue that  @remi-filament pointed out.

Having already wasted enough time on it, I went all in and adjusted the whole thing to follow odoo's example, and I believe the result is much more comprehensive.

Please let me know what you think!

PS: I will update the PR description and screenshots to reflect the new state of things in order to avoid confusion in the future

remi-filament commented:
Thanks @ljmnoonan LGTM it works well for invalid inputs !
From my point of view, this can be merged @legalsylvain

legalsylvain commented:
/ocabot merge minor

Thanks !
